src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/AssessmentEntity.kt

```diff
@@ -20,6 +21,7 @@ import javax.persistence.Table
 interface AssessmentRepository : JpaRepository<AssessmentEntity, UUID> {
   fun findAllByAllocatedToUser_Id(userId: UUID): List<AssessmentEntity>
   fun findByApplication_IdAndReallocatedAtNull(applicationId: UUID): AssessmentEntity
+  fun findByApplication_Id(applicationId: UUID): AssessmentEntity?
```

Comment:
This will throw an exception when more than 1 Assessment exists for an Application which would be the case after a reallocation.  The method above `findByApplication_IdAndReallocatedAtNull` will do what you need here though as reallocating is transactional so there can only ever be one Assessment per Application where reallocatedAt is null.